### DISCUSSION TITLE
http access mxf2raw slow

### DISCUSSION THREAD

**Discussion by donkon** [bbc/bmx#13] [General]
We use mxf2raw in combination with http to retrieve metadata from our MXF files, e.g.

`mxf2raw -i http://webserver/path/to/file.mxf`

For some files this commands takes quite long (1 - 2 minutes), while accessing the same file over a filesystem is quite fast (1 - 2 seconds). I guess, bmx is seeking in the mxf file by sending a bunch of http range requests, which just takes time in sum and we end up in waiting 1 -2 minutes.

Since this long time is a problem for our workflow, is there a way to come around this? We had the same problem with ffprobe and figured out, that setting "seekable=0" solved this problem.

**Comment by philipnbbc** [comment]
I don't think there is currently a workaround to avoid seeking, but it should be easy to add an option to treat the input as not seekable. The changes made recently in https://github.com/bbc/bmx/pull/9 reduced the number of read access and that might help as well - `--http-min-read` was changed to default to 1MB from 64KB.

A commandline option alongside `--http-min-read` could be passed through to the [mxf_http_file_open_read()](https://github.com/bbc/bmx/blob/0683b217572b369e0059de3b56939292c5ca7050/src/common/MXFHTTPFile.cpp#L455) method. I could have a look at implementing that as it would generally be useful to have, particularly for XDCAM files for example that have many partitions. One thing I need to look at is supporting files where the complete header metadata is in the footer partition. That would ideally require a seek to avoid reading the whole file.

**Reply by donkon** [reply]
That would be be great if you could have look at implementing an option to treat the input as not seekable (and taking the footer partition into account). I already tried out the `--http-min-read` option but it did not improve anything :-/ Indeed the problem is related to XDCAM files, which reside in Buckets or Webservers.

N.b. generally the tool does a good job and helps us a lot to build our software for converting german broadcast formats (ARD/ZDF) :+1:

**Comment by philipnbbc** [comment]
I've created a PR that adds some options to speed things up: https://github.com/bbc/bmx/pull/15.

I still need to do a couple of tests but if you could take a look to see if it works for you that would be great.

**Reply by donkon** [reply]
Thx, I will take a look
